src/rdkafka_idempotence.c

```diff
@@ -496,59 +496,77 @@ void rd_kafka_idemp_pid_update(rd_kafka_broker_t *rkb,
 }
 
 
+/**
+ * @brief Requests a new PID and triggers a re-request.
+ *
+ * @locality any
+ * @locks none
+ */
+static void rd_kafka_drain_reset_pid(rd_kafka_t *rk) {
```

Comment:
Thank you for the review!

I thought it would be better to narrow the critical sections used in the `..drain_done()` which includes debug logging, also the minimal nesting without auxiliary variables makes it more easier to understand, however, if it is a fragile code section I can change it in the `drain_done()` txn path directly.

Reply:
Resolved with e4cb5c41e03d9212c0e355e59642bc9e2518d8d6